### ISSUE TITLE
Feature Request - Support Workspaces -w / --workspace=(PATH|PROJECT_NAME) 

### ISSUE THREAD

**Issue by metawrap-dev** [raineorshine/npm-check-updates#1099]
### Feature Request - Support Workspaces -w / --workspace=(PATH|PROJECT_NAME) 

Upgrade all workspaces

`ncu -u --workspace=modules/*`
`ncu -u -w=*`

Upgrade just one workspace,

`ncu -u --workspace=@metawrap/metawrap-core`
`ncu -u --workspace=modules//metawrap-core`
`ncu -u -w=@metawrap/metawrap-core`

See:

https://docs.npmjs.com/cli/v7/using-npm/workspaces#running-commands-in-the-context-of-workspaces

**Comment by raineorshine** [comment]
@metawrap-dev?

**Comment by metawrap-dev** [comment]
> Thanks for the suggestion!
> 
> Is the proposed `ncu -u -w=*` any different than `ncu -u --deep`?

`ncu -u --deep` could affect more dirs than those specified as workspaces. Would be equivalent only when dirs equals workspaces. 

Would also be great to be able to go `ncu -u -w=@metawrap/client-*` to specify a named subset of worspaces. 

> Is the proposed `ncu -u -w=@metawrap/metawrap-core` any different than `ncu -u --packageFile modules/@metawrap/metawrap-core/package.json`?

Would be the same effect, but would be nicer to refer to workspaces instead of folders. 

What is driving this suggestion is that If workspaces are going to be the way of referring to sub-projects across all the node tooling, would be great to support that for consitency. 

> Trying to clarify if this is more of an alias or a new feature.

Thanks! Sorry for the delay. Not in GH as often as I need to be :)

**Comment by rogeriorc** [comment]
You can use npm exec:

```
npm exec --workspaces --include-workspace-root -- ncu -u
npm exec --workspace=@metawrap/metawrap-core -- ncu -u
```

**Comment by ghdoergeloh** [comment]
I would also love to have this option.

An additional bug to have a look at would be the parallel run of `npm install`.

`✔ npm install in each project directory to install new versions? … yes`

If you have two `package.json` with the same dependency both will try to put the package into the root `node_modules` which causes errors like `npm ERR! EEXIST: file already exists`.
So I would ether need an option to disable `npm install` after a ncu execution to run `npm install` manually afterwards or ncu musst check for workspaces and run `npm install` just once in a workspace (preferred in the root directory).

**Comment by raineorshine** [comment]
Please create a separate issue with exact steps to reproduce, including your folder layout and ncu config. 

It is only loosely related to the topic of this issue, which is to add a new option `--workspace [name]` as a shorthand for `--packageFile [path-to-workspace]`.

**Comment by httpete** [comment]
I am finding in my pure npm workspaces monorepo this works just great: I have two scripts at the root package. Once updates the root, and one updates all the workspaces interactively.

```
    "update": "ncu -u -i",
    "update:workspaces": "npm exec --workspaces -- ncu -u -i",

**Comment by raineorshine** [comment]
Workspace support added in prerelease `v16.2.0-0`. Open for feedback :). Your help is much appreciated in finding unidentified edge cases.

The API is slightly different than the OP in order to more closely match npm's `--workspaces` and `--workspace` options.

```
npx npm-check-updates@v16.2.0-0 --workspaces
```

## How it works

Upgrade all workspaces:

```
ncu --workspaces
ncu -ws
```

Upgrade a single workspace:

```
ncu --workspace a
ncu -w a
```

Upgrade more than one workspace:

```
ncu --workspace a --workspace b
ncu -w a -w b
```

Upgrade all workspaces AND the root project:

```
ncu --withWorkspaces
ncu -wws
```

Upgrade a single workspace AND the root project:

```
ncu --withWorkspace a
ncu -ww a
```

## Notes

- If run in `--interactive` mode, ncu will prompt to `npm install` once in the root project rather than separately in each workspace (#1182). 
- Running `--deep` will not trigger workspace support.

**Comment by httpete** [comment]
Seems to work exactly the same for me @raineorshine thanks!

**Comment by raineorshine** [comment]
Published to `latest` on `v16.3.0`. 

Note that `--withWorkspaces` has changed to a cleaner syntax of `--workspaces --root` (and `--withWorkspace` to `--workspace X --root`). Thanks to @mesqueeb for the suggestion.
